components/plurals/src/lib.rs

```diff
@@ -211,26 +211,27 @@ impl PluralCategory {
     ///
     /// let mut categories = PluralCategory::all();
     ///
-    /// assert_eq!(categories.next(), Some(&PluralCategory::Few));
-    /// assert_eq!(categories.next(), Some(&PluralCategory::Many));
-    /// assert_eq!(categories.next(), Some(&PluralCategory::One));
-    /// assert_eq!(categories.next(), Some(&PluralCategory::Other));
-    /// assert_eq!(categories.next(), Some(&PluralCategory::Two));
-    /// assert_eq!(categories.next(), Some(&PluralCategory::Zero));
+    /// assert_eq!(categories.next(), Some(PluralCategory::Zero));
+    /// assert_eq!(categories.next(), Some(PluralCategory::One));
+    /// assert_eq!(categories.next(), Some(PluralCategory::Two));
+    /// assert_eq!(categories.next(), Some(PluralCategory::Few));
+    /// assert_eq!(categories.next(), Some(PluralCategory::Many));
+    /// assert_eq!(categories.next(), Some(PluralCategory::Other));
     /// assert_eq!(categories.next(), None);
     /// ```
     ///
     /// [`Plural Categories`]: PluralCategory
-    pub fn all() -> impl ExactSizeIterator<Item = &'static Self> {
+    pub fn all() -> impl ExactSizeIterator<Item = Self> {
```

Comment:
Thought: I would rather have the order change be in its own PR, but I'm not going to stop it from landing here.